gems/aws-partitions/lib/aws-partitions/endpoint_provider.rb

```diff
@@ -54,27 +60,71 @@ def signing_region(region, service)
           .fetch('region', region)
       end
 
+      # @param [String] region The region used to fetch the partition.
+      # @param [String] service Used only if dualstack is true. Used to find a
+      #   DNS suffix for a specific service.
+      # @param [Hash] variants Endpoint variants such as 'fips' or 'dualstack'
+      # @option variants [Boolean] :dualstack When true, resolve a dualstack
+      #   endpoint.
+      # @option variants [Boolean] :fips When true, resolve a FIPS endpoint.
       # @api private Use the static class methods instead.
-      def dns_suffix_for(region)
-        get_partition(region)['dnsSuffix']
+      def dns_suffix_for(region, service, variants)
+        partition = get_partition(region)
+        partition_variants = partition.fetch('defaults', {})
+                                      .fetch('variants', [])
+        service_variants = partition.fetch('services', {})
+                                    .fetch(service, {})
+                                    .fetch('defaults', {})
+                                    .fetch('variants', [])
+        service_variant = get_variant(service_variants, variants)
+        unless service_variant['dnsSuffix']
+          partition_variant = get_variant(partition_variants, variants)
+        end
+
+        service_variant['dnsSuffix'] ||
+          partition_variant['dnsSuffix'] ||
+          partition['dnsSuffix']
       end
 
       private
 
-      def endpoint_for(region, service, sts_regional_endpoints)
+      def endpoint_for(region, service, sts_endpoint, variants)
+        # partition default endpoint
         partition = get_partition(region)
+        partition_variant = get_variant(
+          partition.fetch('defaults', {}).fetch('variants', []), variants
+        )
+        default_endpoint = if partition_variant.empty?
+          partition['defaults']['hostname']
```

Comment:
Should we be setting a default_endpoint in this case ( and when variants is non empty)?

Reply:
NVM - this will raise later.  But may make sense to move those checks up?